pkg/inventory/go-template.go

```diff
@@ -0,0 +1,70 @@
+package inventory
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+        "text/template"
+)
+
+
+type GoTemplate struct {
+	Template string               `yaml:"template"`
+        Params map[string]interface{} `yaml:"params"`
+}
+
+func (gt *GoTemplate) Process(ns *string, r *Resource) error {
+
+	// Pre-process to convert to absolute pathing
+	abs, err := filepath.Abs(filepath.Join(r.Prefix, gt.Template))
+	if err != nil {
+		return err
+	}
+	gt.Template = abs
+
+	fmt.Printf("Found template %s and one set of params\n", gt.Template)
+        ext := filepath.Ext(gt.Template)
+        basename := baseName(gt.Template, ext)
+        err = processOneGoTemplate(gt.Template, gt.Params, r, basename + ".yml")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func processOneGoTemplate(tpl string, ps map[string]interface{}, r *Resource, outName string) error {
+	if outName == "" {
+		outName = filepath.Base(tpl) + ".yml"
+	}
+
+        t, err := template.ParseFiles(tpl)
+        if err != nil {
+                log.Fatal(err)
+        }
+
+	// write resulting resource to file
+	outputDir := filepath.Join(r.Output, r.Action)
+	out, err := os.Create(filepath.Join(outputDir, outName))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if e := out.Close(); e != nil {
+			err = e
+		}
+	}()
+	log.Printf("wrote %s\n", out.Name())
+        err = t.Execute(out, ps)
+	if err != nil {
+		return err
+	}
```

Comment:
Sooo... i'm starting to think that this process needs to become a helper method that is used commonly by all template processors. We may have need in the future to optimize this in some way, and I would hate to have to do it multiple times. You want to take a crack at that?

Reply:
This isn't a straight copy of the `processOneTemplate` function in `openshift-template.go`. The types and the number of arguments differ so if you think this could/should be refactored to support both I suggest we open a separate issue for this.